src/main/zapHomeFiles/hud/components/AlertDetailsModal.vue

```diff
@@ -0,0 +1,120 @@
+        <tbody>
+          <tr>
+            <td v-t="'message.alerts_field_url'"></td>
+            <td @click="messageSelected(details.messageId)">
+              <a href="#">{{ details['url'] }}</a>
+            </td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_description'"></td>
+            <td>{{ details['description'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_risk'"></td>
+            <td>{{ details['risk'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_confidence'"></td>
+            <td>{{ details['confidence'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_parameter'"></td>
+            <td>{{ details['parameter'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_attack'"></td>
+            <td>{{ details['attack'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_evidence'"></td>
+            <td>{{ details['evidence'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_cweid'"></td>
+            <td>{{ details['cweid'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_wascid'"></td>
+            <td>{{ details['wascid'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_other'"></td>
+            <td>{{ details['other'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_solution'"></td>
+            <td>{{ details['solution'] }}</td>
+          </tr>
+          <tr>
+            <td v-t="'message.alerts_field_reference'"></td>
+            <td v-if="details['reference']">
+              <li v-for="link in details['reference'].split('\n')">
+                <a :href="link" target="_top">{{ link }}</a>
+              </li>
+            </td>
+            <td v-else></td>
+          </tr>
+        </tbody>
+      </table>
+    </div>
+  </NavModal>
+</template>
+
+<script>
+import { EventBus } from "../libs/event-bus.js";
+import NavModal from "./NavModal.vue";
+
+export default {
+  template: "#alert-details-modal-template",
+  props: ["show", "title"],
+  components: {
+    NavModal
+  },
+  methods: {
+    close: function() {
+      this.$emit("close");
+    },
+    messageSelected: function(id) {
+      //TODO: replace tabId, frameId w/ props
+      navigator.serviceWorker.controller.postMessage({
+        tabId: tabId,
+        frameId: frameId,
+        action: "showHttpMessageDetails",
+        tool: "history",
+        id: id
+      });
+    },
+    back: function() {
+      //TODO: replace app (Vue context w/ props)
+      app.keepShowing = true;
+      app.isAlertDetailsModalShown = false;
+      this.port.postMessage({ back: true });
+    }
+  },
+  data() {
+    return {
+      port: null,
+      details: {}
+    };
+  },
+  created() {
+    let self = this;
+
```

Comment:
I haven't tried building the code from your PR yet, but I wonder a little why the clone of 'this'. Shouldn't the fat arrow function of the EventBus pass the scope natively? And even if this is merely an assignment instead of a deep copy, it does cost some cycles. Considering that this is all being run simultaneously in one browser window I would look for all opportunities to shave performance hits.